pkg/registry/client.go

```diff
@@ -145,6 +146,36 @@ func ClientOptCredentialsFile(credentialsFile string) ClientOption {
 	}
 }
 
+func (c *Client) newResolver(insecure, plainHTTP bool) (remotes.Resolver, error) {
+	headers := http.Header{}
+	headers.Set("User-Agent", version.GetUserAgent())
+	opts := []auth.ResolverOption{auth.WithResolverHeaders(headers)}
+
+	if insecure {
+		httpClient := http.DefaultClient
+		httpClient.Transport = &http.Transport{
+			TLSClientConfig: &tls.Config{
+				InsecureSkipVerify: true,
+			},
+		}
+		opts = append(opts, auth.WithResolverClient(httpClient))
+	}
+	if plainHTTP {
+		opts = append(opts, auth.WithResolverPlainHTTP())
+	}
+
+	return c.authorizer.ResolverWithOpts(opts...)
+}
+
+func (c *Client) WithResolver(insecure, plainHTTP bool) error {
```

Comment:
I'm coming across this PR as I'm implementing [HTTP support for OCI in Flux right now](fluxcd/source-controller#807) and I'd love to see it progressing. Thanks for coming up with this change.

Wouldn't this be more in line with how all other options of a client are configured if this was a `ClientOption`?

Reply:
Yep, I do agree, I prefer that `ClientOption` approach as well ([more info](https://github.com/helm/helm/pull/10638#issuecomment-1059426505)). In fact, my PR got stuck for months and I ended up creating [a new one](https://github.com/helm/helm/pull/11129) just performing this simple change. I really hope the maintainers can move it forward soon.

We need that functionality in [Kubeapps](kubeapps.dev) as we are currently using the Helm pre 3.8 code for pulling OCI artifacts and, without customizing the resolver... we can't use it as we need.
BTW, looking forward to getting HTTP support in Flux, we will give it a try in Kubeapps soon :P

Reply:
Like @antgamdia said,  you can see new pr,  or this commit that base on @antgamdia pr, https://github.com/helm/helm/pull/10638#issuecomment-1107806745. 
I also hope the manitainers can move these pr soon, because i use harbor self-sign OCI registry.

Reply:
Hu, that's funny as I also created [a PR](#11139), just that it only adds the specific `ClientOption` for mandating plain HTTP. We'll have to figure out the desired API and machinery.